Algorithm/QCAlgorithm.History.cs

```diff
@@ -479,7 +479,7 @@ public BaseData GetLastKnownPrice(Security security)
                 security.Symbol,
                 resolution,
                 security.Exchange.Hours,
-                MarketHoursDatabase.FromDataFolder().GetDataTimeZone(security.Symbol.ID.Market, security.Symbol, security.Symbol.SecurityType),
```

Comment:
We're not guaranteed that the data time zone and exchange time zone are the same. We can do something like this:
```csharp
var dataTimeZone = security.Subscriptions.Select(config => config.DataTimeZone).First();
```

Another possibility here is when we do `MarketHoursDatabase.GetEntry` and don't find an entry **and** an `overrideTimeZone` was specified, add the entry to the dictionary. This would guarantee that we're always fetching the configured overriden time zone.

@jaredbroad @StefanoRaggi -- Thoughts on adding entries to market hours database for custom data (would happen as part of `AddData<T>`)